### DISCUSSION TITLE
Option to output numbers in scientific notation?

### DISCUSSION THREAD

**Discussion by loops** [wxMaxima-developers/wxmaxima#1582] [Q&A]
Hi,

Did my best searching for an answer on this.   Would like wxMaxima to leave numbers in the  scientific format that is used in Maxima itself (for example:  7.4e+6) .    Currently wxMaxima is converting such numbers to  7.4 * 10^6 in output which I find gets in my way both in comprehension and when copying-and-pasting.

Is this currently an option that can be set?  

Thanks for the great program, i'm really enjoying it!
Sean

**Comment by gunterkoenigsmann** [comment]
```
load("engineering-format");
engineering_format_min:.01$
engineering_format_max:1000$
fpprintprec:6$
```

**Reply by loops** [reply]
Hi gunterkoenigsmann,

I should have mentioned discovering and trying the "engineering-format" package already.  But it doesn't seem to change the output in wxMaxima in the way I hoped.  When say "32e7" is entered in wxMaxima it is displayed as "320.0 10^6"  with a space in between the base and the ten-exponent.   Instead I'd like to see the number exactly as it is displayed in the command-line version of Maxima, that is:  "320.0e+6".

Thanks so much.

**Comment by daute** [comment]
You could use Maxima's `printf()`command to output variable values, e.g.

```
a:32e7$
printf(false, "~e", a);
```
The output would be:
3.2E+8
(The scientific value is normalized, so that there is exactly one digit before the comma).

Wolfgang

**Reply by loops** [reply]
Hi daute,

Thank you for the idea.  But it seems onerous to have to wrap every output in such a procedure.  And I'm not sure how to print such a result in a greater expression, say in the numerator above a constant for example.  It seems like if what I wanted was possible you or gunterkoenigsmann would have been able to point me to it, since you're both maintainers.   I will just accept that wxMaxima always displays exponent results in a consistent fashion and there is no option to use the same format that Maxima itself chooses. 

Thanks again,
Sean

**Reply by gunterkoenigsmann** [reply]
We definitely need a lisp guy in the wxMaxima team: many of the current bug reports report bugs in parts of wxmathml.lisp I don't understand.
